src/geometry/dual_quaternion_ops.rs

```diff
@@ -0,0 +1,29 @@
+/*
+ * This file provides:
+ * NOTE: Work in progress https://github.com/dimforge/nalgebra/issues/487
+ *
+ * References:
+ *   Multiplication:
+ *   - https://cs.gmu.edu/~jmlien/teaching/cs451/uploads/Main/dual-quaternion.pdf
+ * ===================
+ *
+ */
+
+use std::ops::{Mul};
+use crate::{DualQuaternion, Scalar, DefaultAllocator, U4, U1, SimdRealField};
+use simba::simd::SimdValue;
+use crate::base::allocator::Allocator;
+
+impl <N: Scalar + SimdValue + SimdRealField> Mul<DualQuaternion<N>> for DualQuaternion<N>
```

Comment:
I added the `Scalar + SimdValue` constraint so that I could leverage the existing `impl Mul for Quaternion` (which consumes self, naturally).

Otherwise I could get a borrow checker error from trying to use `self.rot` below twice.

I'm not too familiar with all of the traits in nalgebra. Is there a better way that I should handle this?

Reply:
This is odd. The `SimdRealField` bound should already imply `Scalar + SimdValue` so adding these two should not have made any difference.

Reply:
Mmmm I removed them and it worked fine. I must've just been missing the `N::Element: SimdRealField` bounds when I first got that error or something.
This works now thanks.